### ISSUE TITLE
AVAROK_DISABLE_WATCHDOGS still leaves three watchdogs armed (and AVAROK_SIMHASH_LOOP is parsed loosely)

### ISSUE THREAD

**Issue by tbraun96** [Avarok-Cybersecurity/atlas#1204]
`AVAROK_DISABLE_WATCHDOGS=1` is documented, and logged at boot, as disabling the
auto-watchdogs. #1199 gates the API-layer token-loop watchdog on it — the site
issue #1135 reported. A sweep of the whole serving path found that **three more
watchdogs still fire** with the flag set, so the documented promise stays false
after #1199 lands. Citations are against `main` at `546aa85b2b`.

## Still armed with `AVAROK_DISABLE_WATCHDOGS=1`

| # | watchdog | gate site | firing site | why it escapes |
|---|---|---|---|---|
| 1 | think-skip hard-stop | — | `crates/spark-server/src/scheduler/decode_logits_step.rs:413-421`; MTP twin `scheduler/emit_step.rs:111-116` | no `disable_watchdogs` check in either block. Sets `a.finished = true`, `a.guard_stop = Some(GUARD_STOP_THINK_SKIP)` after 50 consecutive stray `</think>` |
| 2 | fuzzy-repetition | `decode_logits_step.rs:954` | `decode_logits_step.rs:979-982` | gated **only** on the `SchedLevers::loop_watchdog` atomic. The content-loop watchdog a few hundred lines away checks `loop_watchdog()` **and** `disable_watchdogs`; this one checks only the first |
| 3 | SimHash semantic-loop | `api/chat_stream/handle_token.rs:665` | `handle_token.rs:689-706` | gated only on its own `AVAROK_SIMHASH_LOOP` |

For contrast, the watchdogs that **are** correctly gated (all via
`!levers.disable_watchdogs` / `!ctx.sampling.disable_watchdogs`): content-loop
(`decode_logits_content.rs:150`→`:202`, twin `emit_step.rs:337`→`:358`),
inter-tool prose budget (`:221`→`:253`, twin `:383`→`:396`), post-think content
cap F1 (`:100`→`:109`, twin `:321`→`:330`), thinking-loop
(`decode_logits_step.rs:506`→`:517`), F2 confidence early-stop
(`logit_processors/f2_confidence.rs:32`→`:48`), mid-word `</think>` defer
(`logit_processors/mid_word.rs:28`→`:37`).

## Two findings in the same family

**4. `AVAROK_SIMHASH_LOOP` is parsed loosely.**
`crates/spark-server/src/api/chat_stream/handle_token.rs:23-26`:

```rust
*ON.get_or_init(|| std::env::var("AVAROK_SIMHASH_LOOP").as_deref() != Ok("0"))
```

Only the exact string `"0"` turns it off. `AVAROK_SIMHASH_LOOP=false` reads as
**ON** — the same defect #1200 fixed for `AVAROK_PREFIX_SUBBLOCK`. The doc
comment's claim that "`=0` disables" is true but incomplete. Note this sits in
the very file #1199 edits, and is inconsistent with `parse_disable_watchdogs`
in the same crate, which is strict.

**5. The `disable_watchdogs` rule has two implementations.**
`crates/spark-server/src/scheduler/helpers.rs:197-204` is the SSOT:

```rust
pub(crate) fn parse_disable_watchdogs(env: Option<&str>) -> bool {
    match env {
        Some(v) => { let v = v.trim(); v == "1" || v.eq_ignore_ascii_case("true") }
        None => false,
    }
}
```

`crates/spark-model/src/ssm_reserve/decode_ring.rs:152-158` re-spells it as
`watchdogs_disabled_from_value` because `spark-model` cannot depend on
`spark-server`. Both doc comments (`decode_ring.rs:49-51`, `ssm_reserve.rs:13`)
flag it as a deliberate mirror. They agree **today**; nothing enforces that they
keep agreeing, which is the "one flag, two implementations" class that has bitten
four Atlas levers before.

## Proposed work (its own PR, its own campaign)

1. Gate 1–3 on `disable_watchdogs`, matching the content-loop watchdog's
   `loop_watchdog() && !disable_watchdogs` shape.
2. Make `simhash_loop_enabled()` strict, with the same polarity as
   `parse_disable_watchdogs`, and a defect test plus a control.
3. Pin the two parsers in sync — a table-driven test over the same input vector
   asserting both implementations agree, so a future edit to one goes red.

Deliberately **not** bolted onto the #1199/#1200 campaign (#1203): four call
sites across two crates is a second feature-sized change, and same-night
production code composed into an unrelated certification run is what produced 13
composition-only CI failures in campaign 3.

**#1135 should stay open** until 1–3 land; #1199 covers one of four sites.

**Comment by tbraun96** [comment]
**A trap in item 4 that would turn the specified fix into an outage.**

The issue says to make `AVAROK_SIMHASH_LOOP`'s parse strict, "with the same
polarity as `parse_disable_watchdogs`". Read literally that is wrong, and the
wrong version is silent.

The two flags have **opposite defaults**:

| flag | parser | absent means |
|---|---|---|
| `AVAROK_DISABLE_WATCHDOGS` | `parse_disable_watchdogs` (`helpers.rs:197`) | `None => false` — watchdogs **ON** |
| `AVAROK_SIMHASH_LOOP` | `simhash_loop_enabled` (`handle_token.rs:23-26`) | `!= Ok("0")` — the watchdog is **ON** |

`disable_watchdogs` is a *disable* switch defaulting to not-disabled;
`simhash_loop` is an *enable* switch defaulting to enabled. Both currently leave
the watchdog armed when unset, and both must keep doing so — but a
copy-paste of `parse_disable_watchdogs` into `simhash_loop_enabled` yields
`None => false`, which **silently disarms the SimHash semantic-loop watchdog on
every deployment that does not set the variable**, i.e. all of them. Its own doc
comment says "Default ON" and explains why an operator would ever turn it off.

The correct strict form keeps default-ON:

```rust
match std::env::var("AVAROK_SIMHASH_LOOP").ok().as_deref().map(str::trim) {
    None => true,                                   // default ON, unchanged
    Some(v) if v == "1" || v.eq_ignore_ascii_case("true")  => true,
    Some(v) if v == "0" || v.eq_ignore_ascii_case("false") => false,
    Some(other) => { /* refuse or warn — today `off`, `no`, a typo all read ON */ }
}
```

The control that proves it: `AVAROK_SIMHASH_LOOP` **unset** must still enable the
watchdog. That is the assertion a naive port fails, and it is not the one
someone writing "make the parse strict" would think to add.

## Anchors for the other three, so the edit is mechanical

| item | file:line (on `d874944b33`) | shape today |
|---|---|---|
| think-skip hard-stop | `scheduler/decode_logits_step.rs:416` | `if a.think_skip_count >= 50 { a.finished = true; … }` — no gate at all |
| fuzzy-repetition | `scheduler/decode_logits_step.rs:959` | `if sched.levers.loop_watchdog() && !a.finished` — missing `&& !…disable_watchdogs` |
| SimHash semantic-loop | `api/chat_stream/handle_token.rs:665` | gated only on `simhash_loop_enabled()` |

Plus both MTP twins in `emit_step.rs` (`:111-116` for think-skip), which the
content-loop watchdog already treats as a pair.

This is production code in `crates/spark-server`, so it needs its own
certification campaign — which was the agreed disposition when #1199 landed:
fix these for real, on their own campaign, not smuggled into one built to
certify something else.
